### ISSUE TITLE
Truncating a struct can be very slow (minutes)

### ISSUE THREAD

**Issue by sliedes** [NationalSecurityAgency/ghidra#6504]
**Describe the bug**

I am working on analyzing an x86_64 ELF binary that is less than 1 MiB in size (just in case these details are relevant).

The binary that I'm analyzing seems to have very large structures, more than a megabyte (if relevant, they seem to be classes, as in the `*this` struct in C++).

Working in the structure editor, expanding the size of a struct by editing the `Size` field works instantaneously. However, when I immediately after expanding a struct do the reverse and truncate it (say, from 0x350000 to 0x150000), the Ghidra UI _gets stuck for several minutes_, occupying at times 8-16 threads (apparently for GC; see analysis below), and at other times only one.

**To Reproduce**

1. Edit a struct type that is used in the program
2. Set size to 0x350000
3. Set size to 0x150000
4. Acknowledge the warning about truncation

**Expected behavior**

The struct is truncated, UI does not get struck for several minutes.

**More detailed analysis**

I ran Ghidra under perf to see where the time is spent. I'm not sure if I did everything the easiest way, since this is my first time using perf on Java code:

```sh
$ SCRIPT_FILE=/opt/ghidra/ghidraRun SCRIPT_DIR=/opt/ghidra _JAVA_OPTIONS="-Dawt.useSystemAAFontSettings=lcd -XX:+UnlockDiagnosticVMOptions -XX:+DumpPerfMapAtExit" perf record -g bash +x /opt/ghidra/support/launch.sh fg jdk Ghidra '' '' ghidra.GhidraRun
[... exercise Ghidra, observe that the CPU-consuming Ghidra PID is 11395579; then quit Ghidra.]
$ perf report -g
```

Here's what I see at the top (no one single hotspot :():

Samples: 8M of event 'cycles:P', Event count (approx.): 5462731961825
| Children  | Self  | Command         |    Shared Object   | Symbol    |
| --------- | ----- | --------------- | ------------------ | --------- |
| 11.79%    | 0.05% | AWT-EventQueue- | [JIT] tid 1139579  | [.] java.security.AccessControlContext java.security.AccessController.getStackAccessControlContext() |
| 11.32%    | 1.97% | AWT-EventQueue- | libjvm.so          | [.] JVM_GetStackAccessControlContext |
|  8.39%    | 0.00% | AWT-EventQueue- | [unknown]          | [.] 0xffe005442334e378 |
|  6.17%    | 0.26% | AWT-EventQueue- | [JIT] tid 1139579  | [.] void java.awt.EventQueue.invokeLater(java.lang.Runnable) |
|  3.86%    | 2.07% | AWT-EventQueue- | [JIT] tid 1139579  | [.] ghidra.program.model.data.DataType ghidra.program.database.data.DataTypeManagerDB.getBuiltInDataType(long, db.DBRecord) |
|  3.30%    | 0.46% | AWT-EventQueue- | [JIT] tid 1139579  | [.] void java.awt.EventDispatchThread.pumpOneEventForFilters(int) |
|  2.76%    | 0.08% | AWT-EventQueue- | [JIT] tid 1139579  | [.] void ghidra.program.database.data.DataTypeManagerDB.sourceArchiveChanged(ghidra.util.UniversalID) |
|  2.63%    | 1.02% | AWT-EventQueue- | [JIT] tid 1139579  | [.] ghidra.program.database.DatabaseObject ghidra.program.database.DBObjectCache.get(long) |
|  2.56%    | 1.47% | AWT-EventQueue- | [JIT] tid 1139579  | [.] db.Field[] db.Table.findRecords(db.Field, int) |
|  2.48%    | 2.08% | AWT-EventQueue- | [JIT] tid 1139579  | [.] void javax.swing.JTable.clearSelectionAndLeadAnchor() |
|  2.20%    | 1.11% | AWT-EventQueue- | [JIT] tid 1139579  | [.] boolean ghidra.app.plugin.core.compositeeditor.CompositeEditorModel.hasSameSettings(ghidra.program.model.data.DataTypeComponent, ghidra.program.model.data.DataTypeComponent) |

... and a ton of GC Threads at 1.90 %.

**Environment (please complete the following information):**

 - OS: Arch Linux
 - Java Version: OpenJDK 22
 - Ghidra Version: 11.0.3
 - Ghidra Origin: Arch Ghidra package, 11.0.3-1

**Comment by ghidra1** [comment]
This is a known limitation of the structure editor and is a duplicate of previous iissues submitted.  Unfortunately, this performance concern is inherent to the manner in which undefined components are tracked for non-packed structures.

**Comment by ghidra1** [comment]
I am in the process of addressing this concern

**Comment by ghidra1** [comment]
Could you please verify that your use case slowdown has been resolved with the Ghidra 11.2 release.  Thanks
